### PR TITLE
Add NOUNDEFINED variable

### PR DESCRIPTION
**Pull request by amontoison**
@dpo @nimgould

### CODE REVIEW

**Review by nimgould — CHANGES_REQUESTED** [review]
Could you remove the -Wl, in the NOUNDEFINED for linux and mac. The -Wl will be assumed. Thank you!

**Comment by amontoison** [comment]
Is it not an issue for the linker if we just have `Wl,` for platforms where `NOUNDEFINED=""`?

**Comment by nimgould** [comment]
Yes, presumably it is. In that case, we need to put something in for each NOUNDEFINED. We could follow what we have for the LOADALL & LOADNONE. Or do you think that instead we should have the -Wl, for these too? I'm easy, it will simply mean a minor change to the create*shared scripts

**Comment by amontoison** [comment]
I think that the windows linker doesn't have a value for `NOUNDEFINED`.
This is the default and unique behaviour so I suppose that they don't provide a linker option for it.
I propose to keep `-Wl` such that we don't use a wrong option for linkers that we don't know well.

**Comment by nimgould** [comment]
OK, that is fine, could you then add -Wl, to the LOADALL & LOADNONE variables. I believe that @dpo  said that NOADNONE on Macs was not needed, so LOADNONE can be blank there

**Comment by amontoison** [comment]
Yes, I will open another PR for `LOADALL` and `LOADNONE`.

File: system.bb [hunk 1]
```diff
@@ -39,5 +39,6 @@ RANLIB="ranlib"
 
 GETCPU="un"
 DLEXT="$dlext"
-LOADALL="$(flagon -Wl,--whole-archive)"
-LOADNONE="$(flagon -Wl,--no-whole-archive)"
+LOADALL="$(flagon --whole-archive)"
+LOADNONE="$(flagon --no-whole-archive)"
```
Comment (amontoison) [inline system.bb]:
```suggestion
LOADALL="$(flagon -Wl,--whole-archive)"
LOADNONE="$(flagon -Wl,--no-whole-archive)"
```

**Review by nimgould — APPROVED** [review]
OK

**Comment by nimgould** [comment]
All ok now, there was a cycle of approve and please reapproves that got into a knot

**Pull request merged**